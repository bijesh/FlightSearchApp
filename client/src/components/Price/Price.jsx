import React from 'react';
import BpkButton from 'bpk-component-button';
import ITINERARY_STYLES from './Price.scss';

const priceStyles = className => ITINERARY_STYLES[className] || 'UNKNOWN';

const Price = React.memo(({ bookingDetails }) => (
         <div className={priceStyles('Price__footer')}>
            <div className={priceStyles('Price__airline')}>
              <div className={priceStyles('Price__airline__price')}>Â£{bookingDetails.price}</div>
              <div className={priceStyles('Price__airline__name')}>{bookingDetails.agent}</div>
            </div>
            <div className={priceStyles('Price__button')}>
              <BpkButton className={priceStyles('Price__airline__select')}large>Select</BpkButton>
            </div>
          </div>
));
export default Price;