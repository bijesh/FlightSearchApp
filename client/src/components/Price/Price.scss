@import '~bpk-mixins';

.Price {
   
    &__footer {
        margin-top: 8%;
        float: left;
        width: 100%;
    }
    &__button {
        width: 50%;
        float: right;
        margin-top: 5%;
    }
    &__airline {
            width: 50%;
            float: left;
        &__price {
            font-size: 27pt;
            font-weight: bold;
        }
        &__name {
            font-size: 15pt;
            color: #ADADAD;
            margin-top: 8%
        }
        &__select {
            float: right
        }
    }
}