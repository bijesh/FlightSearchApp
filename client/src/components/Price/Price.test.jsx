import React from 'react';
import TestRenderer from 'react-test-renderer';
import Price from './Price';

const PriceDate={
price:100,
agent: "BA"
};

describe('Price', () => {
  it('should render correctly', () => {
    const tree = TestRenderer.create(<Price bookingDetails={PriceDate} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

