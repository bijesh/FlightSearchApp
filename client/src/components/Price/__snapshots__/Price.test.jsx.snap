// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price should render correctly 1`] = `
<div
  className="Price__footer"
>
  <div
    className="Price__airline"
  >
    <div
      className="Price__airline__price"
    >
      Â£
      100
    </div>
    <div
      className="Price__airline__name"
    >
      BA
    </div>
  </div>
  <div
    className="Price__button"
  >
    <button
      className="bpk-button bpk-button--large Price__airline__select"
      disabled={false}
      onClick={null}
      type="button"
    >
      Select
    </button>
  </div>
</div>
`;
